### ISSUE TITLE
Use locally installed library types, not vendored stubs?

### ISSUE THREAD

**Issue by NickCrews** [microsoft/pylance-release#2444]
Hi! I think my issue is similar to https://github.com/microsoft/pylance-release/issues/1197, but a little different??

I have cloned the pandas library (https://github.com/pandas-dev/pandas) into a directory at `/workspaces/pandas/`, and set up a dev environment following their instructions. This includes installing the library as editable (eg `pip install -e .`). Now, my python interpreter finds the local version of the source code at `/workspaces/pandas/pandas`, which is great.

However, pylance is not using this local copy for type hints. It is using the stubs under (I'm in a devcontainer in docker, ignore the root weirdness) `/root/.vscode-server/extensions/ms-python.vscode-pylance-2022.3.0/dist/bundled/stubs/pandas/`. The problem is that as I pull in changes from upstream pandas, the types used in pandas change, but pylance keeps looking at the stale, vendored stubs.

My workaround right now is I just symlinked the fresh copy to where pylance goes looking, so pylance actually uses my fresh version (eg `ln -s /workspaces/pandas/pandas /root/.vscode-server/extensions/ms-python.vscode-pylance-2022.3.0/dist/bundled/stubs/pandas`. But, this isn't ideal.

pandas doesn't currently have a `py.typed` file, IDK if that is important.

Is this a bug? Do I need to change configuration somewhere? Let me know if this doesn't make sense, it is confusing and I probably didn't explain it super well. Thank you!

**Comment by erictraut** [comment]
I think you've already identified the problem. The current pandas library doesn't have a "py.typed" file, so type checkers like pyright will look for type information elsewhere before falling back on inferring types from the code. This is the behavior dictated by PEP 561. You should be able to solve the problem by adding an empty "py.typed" file to the root of the pandas directory.

**Comment by judej** [comment]
Please reopen this issue if the fix suggested above does not work. Thanks

**Comment by NickCrews** [comment]
Hi! So I just did a fresh clone and dev install of pandas, trying to reproduce. After cloning, I added a `py.typed` file manually before doing the `pip install` steps (https://pandas.pydata.org/docs/dev/development/contributing_environment.html#creating-a-python-environment). I assumed that then the installed version of pandas would have the py.typed, and pylance would pick it up.

Now it does something even weirder? watch the screencapture below. On some of the old API, such as ArrayLike, it picks up the symbol and can follow it to the definition, in the locally installed dev folder. But, on the newer API (ie that isn't yet in the stubs that are vendored with pylance) such as NDFrameT, pylance can't resolve the symbol, even though that symbol is actually defined in the exact same `_typing.py` file where `ArrayLike` is!

![Kapture 2022-03-08 at 17 29 08](https://user-images.githubusercontent.com/10820686/157349346-943265bc-e9d3-42c3-9a88-d29c59a3d8f7.gif)

Is there some other set of steps I need to be doing for pylance to pick up on the py.typed file? reset some cache? I reloaded the vscode window a bunch.
